avm/res/api-management/service/tests/e2e/defaults/main.test.bicep

```diff
@@ -36,13 +36,23 @@ resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
 // ============== //
 
 @batchSize(1)
-module testDeployment '../../../main.bicep' = [for iteration in [ 'init', 'idem' ]: {
-  scope: resourceGroup
-  name: '${uniqueString(deployment().name, resourceLocation)}-test-${serviceShort}-${iteration}'
-  params: {
-    name: '${namePrefix}${serviceShort}001'
-    location: resourceLocation
-    publisherEmail: 'apimgmt-noreply@mail.windowsazure.com'
-    publisherName: '${namePrefix}-az-amorg-x-001'
+module testDeployment '../../../main.bicep' = [
+  for iteration in ['init', 'idem']: {
+    scope: resourceGroup
+    name: '${uniqueString(deployment().name, resourceLocation)}-test-${serviceShort}-${iteration}'
+    params: {
+      name: '${namePrefix}${serviceShort}001'
+      location: resourceLocation
+      publisherEmail: 'apimgmt-noreply@mail.windowsazure.com'
+      publisherName: '${namePrefix}-az-amorg-x-001'
+      managedIdentities: {
+        systemAssigned: true
+      }
+      tags: {
```

Comment:
A default test should only include required parameters. Optional parameters should be removed.
If failing WAF, we should evaluate one of the following:
- remove the parameter from the test and add the corresponding failing rule to the min suppression group (if e.g. involving an external dependency) 
- keep the parameter in the test and make the corresponding parameter type as required
- remove the parameter from the test, keep its parameter type Optional, but add a default value satisfying the WAF condition

Reply:
updated code to use more logic for setting managed identity. It's now completely optional and will default to a system managed identity unless explicitly overridden.